zend_abstract_interface/zai_compat.h

```diff
@@ -1,17 +1,20 @@
 #ifndef ZAI_COMPAT_H
 #define ZAI_COMPAT_H
 
-// macros to be used only within ZAI headers or tests to avoid duplication of code lines
 #if PHP_VERSION_ID >= 70000
 #define ZAI_TSRMLS_C
 #define ZAI_TSRMLS_CC
 #define ZAI_TSRMLS_D
 #define ZAI_TSRMLS_DC
+#define smart_str_length(str) (str)->s->len
```

Comment:
I don't love this, and neither does bob. But the point here is to keep seems consistent, the value of that is reduced if the consumer of any API must ifdef around the call site.

Should we decide to use smart_str more widely than we do now, maybe a zai_smart_str is a path we could follow.

This looks like the bear minimum to keep seams tidy.